src/fsharp/MethodCalls.fs

```diff
@@ -1895,8 +1898,36 @@ let GenWitnessExpr amap g m (traitInfo: TraitConstraintInfo) argExprs =
         else 
             Some (mkAnonRecdFieldGet g (anonInfo, argExprs.[0], tinst, i, m))
 
-    | Choice4Of5 expr ->
+    | Choice4Of5 expr -> 
         Some (MakeApplicationAndBetaReduce g (expr, tyOfExpr g expr, [], argExprs, m))
 
-    | Choice5Of5 () ->
-        None
+    | Choice5Of5 () -> 
+        match traitInfo.Solution with 
+        | None -> None // the trait has been generalized
+        | Some _-> 
+        // For these operators, the witness is just a call to the coresponding FSharp.Core operator
+        match g.tryMakeOperatorAsBuiltInWitnessInfo isStringTy isArrayTy traitInfo argExprs with
+        | Some (info, tyargs, actualArgExprs) -> 
+            tryMkCallCoreFunctionAsBuiltInWitness g info tyargs actualArgExprs m
+        | None -> 
+            // For all other built-in operators, the witness is a call to the coresponding BuiltInWitnesses operator
+            // These are called as F# methods not F# functions
+            tryMkCallBuiltInWitness g traitInfo argExprs m
+        
+/// Generate a lambda expression for the given solved trait.
+let GenWitnessExprLambda amap g m (traitInfo: TraitConstraintInfo) =
+    let witnessInfo = traitInfo.TraitKey
+    let argtysl = GenWitnessArgTys g witnessInfo
+    let vse = argtysl |> List.mapiSquared (fun i j ty -> mkCompGenLocal m ("arg" + string i + "_" + string j) ty) 
+    let vsl = List.mapSquared fst vse
+    match GenWitnessExpr amap g m traitInfo (List.concat (List.mapSquared snd vse)) with 
+    | Some expr -> 
+        Choice2Of2 (mkMemberLambdas m [] None None vsl (expr, tyOfExpr g expr))
+    | None -> 
+        Choice1Of2 witnessInfo
+        //assert ("A constraint witness could not be found for a built-in constraint solution" |> ignore; false)
```

Comment:
nit: Two lines of commented code should be removed.

Reply:
thanks!